pilot/pkg/xds/ads.go

```diff
@@ -917,80 +917,63 @@ func (conn *Connection) send(res *discovery.DiscoveryResponse) error {
 
 // nolint
 func (conn *Connection) NonceAcked(typeUrl string) string {
-	conn.proxy.RLock()
-	defer conn.proxy.RUnlock()
-	if conn.proxy.WatchedResources != nil && conn.proxy.WatchedResources[typeUrl] != nil {
-		return conn.proxy.WatchedResources[typeUrl].NonceAcked
+	wr := conn.proxy.GetWatchedResource(typeUrl)
+	if wr != nil {
+		return wr.NonceAcked
 	}
 	return ""
 }
 
 // nolint
 func (conn *Connection) NonceSent(typeUrl string) string {
-	conn.proxy.RLock()
-	defer conn.proxy.RUnlock()
-	if conn.proxy.WatchedResources != nil && conn.proxy.WatchedResources[typeUrl] != nil {
-		return conn.proxy.WatchedResources[typeUrl].NonceSent
+	wr := conn.proxy.GetWatchedResource(typeUrl)
+	if wr != nil {
+		return wr.NonceSent
 	}
 	return ""
 }
 
 func (conn *Connection) Clusters() []string {
-	conn.proxy.RLock()
-	defer conn.proxy.RUnlock()
-	if conn.proxy.WatchedResources != nil && conn.proxy.WatchedResources[v3.EndpointType] != nil {
-		return conn.proxy.WatchedResources[v3.EndpointType].ResourceNames
+	wr := conn.proxy.GetWatchedResource(v3.EndpointType)
+	if wr != nil {
+		return wr.ResourceNames
 	}
 	return []string{}
 }
 
 func (conn *Connection) Routes() []string {
-	conn.proxy.RLock()
-	defer conn.proxy.RUnlock()
-	if conn.proxy.WatchedResources != nil && conn.proxy.WatchedResources[v3.RouteType] != nil {
-		return conn.proxy.WatchedResources[v3.RouteType].ResourceNames
+	wr := conn.proxy.GetWatchedResource(v3.RouteType)
+	if wr != nil {
+		return wr.ResourceNames
 	}
 	return []string{}
 }
 
 // nolint
 func (conn *Connection) Watching(typeUrl string) bool {
-	conn.proxy.RLock()
-	defer conn.proxy.RUnlock()
-	if conn.proxy.WatchedResources != nil && conn.proxy.WatchedResources[typeUrl] != nil {
+	wr := conn.proxy.GetWatchedResource(v3.RouteType)
```

Comment:
```suggestion
	wr := conn.proxy.GetWatchedResource(typeUrl)
```

Reply:
This function is legacy, can be removed. So this is why no test failed

Reply:
Oh thats good, I was concerned no tests failed